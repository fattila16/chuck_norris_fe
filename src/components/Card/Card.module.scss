@import 'src/assets/scss/color-palette';
@import 'src/assets/scss/vars';

.card {
    display: flex;
    flex-direction: row;
    flex-shrink: 0;
    align-items: center;
    text-decoration: none;
    border-radius:5px;
    max-height: 200px;
    max-width: 800px;
    width: 100%;
    box-shadow:0px 4px 18px rgba(0,0,0,0.1);
    background-color: white;
    transition: all 0.2s ease;
    padding: 20px;
    &:hover {
        transform: translateY(-1px);
        box-shadow:0px 4px 18px rgba(0,0,0,0.2);
    }
}

.image {
    width: 10%;
    height: 60%;
    border-radius: 5px 5px 0 0;
    object-fit: contain;
    display: block;
}

.fact {
    font-size: 16px;
    padding: 0px 18px 18px 18px;
    overflow-x: hidden;
}

.ul {
    list-style-type: none;
    align-self: start;
    display: flex;
    padding: 0;
    margin: 0;
    span {
        padding-right: 10px;
    }
}

@media only screen and (max-width: $mobile-breakpoint) {
    .fact {
        font-size: 16px;
    }
    
    .image {
        width: 20%;
    }
    .card {
        padding: 20px;
        max-height: none;
        max-width: none;
    }
}