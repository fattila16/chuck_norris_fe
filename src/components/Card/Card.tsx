import React from 'react';
import css from './Card.module.scss';

interface CardProps {
    img: string;
    fact: string;
    categories: string[];
}

const Card: React.FC<CardProps> = (props: CardProps) => {
    const { img, fact, categories } = props;
    return (
        <div className={css.card}>
            <img className={css.image} src={img} alt="chuck-fact-card" />
            <ul className={css.ul}>
                <span>Categories:</span>
                {categories.map((category) => (
                    <li key={category}>{category}</li>
                ))}
            </ul>
            <div className={css.fact}>{fact}</div>
        </div>
    );
};

export default Card;
