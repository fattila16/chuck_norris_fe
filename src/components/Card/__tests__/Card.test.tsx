import React from 'react';
import { shallow } from 'enzyme';
import Card from '../Card';

describe('[Card]', () => {
    it('Should render Card component without error', () => {
        const component = shallow(<Card img="" fact="Test" categories={['testCategory']} />);
        expect(component.getElements()).toMatchSnapshot();
    });
});
