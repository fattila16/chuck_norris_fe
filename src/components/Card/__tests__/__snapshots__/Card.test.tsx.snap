// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Card] Should render Card component without error 1`] = `
Array [
  <div
    className="card"
  >
    <img
      alt="chuck-fact-card"
      className="image"
      src=""
    />
    <div
      className="fact"
    >
      Test
    </div>
  </div>,
]
`;
